File: test/extensions/language/identifiers/cpp/StructureNameClashingWithJavaTest.cpp
#include "gtest/gtest.h"
#include "identifiers/structure_name_clashing_with_java/StructureNameClashingWithJava.h"
#include "zserio/BitStreamReader.h"
#include "zserio/BitStreamWriter.h"

namespace identifiers
{
namespace structure_name_clashing_with_java
{

class StructureNameClashingWithJavaTest : public ::testing::Test
{
protected:
    static const size_t BIT_SIZE;
};

const size_t StructureNameClashingWithJavaTest::BIT_SIZE = 8 * 1 + // all auto optionals
        8 + // Byte
        16 + // Short
        32 + // Integer
        64 + // Long
        64 + // BigInteger
        32 + // Float
        64 + // Double
        8; // String '\0'

TEST_F(StructureNameClashingWithJavaTest, emptyConstructor)
{
    {
        StructureNameClashingWithJava structureNameClashingWithJava;
        ASSERT_FALSE(structureNameClashingWithJava.getByteField().isValueUsed());
        ASSERT_FALSE(structureNameClashingWithJava.getShortField().isValueUsed());
        ASSERT_FALSE(structureNameClashingWithJava.getIntegerField().isValueUsed());
        ASSERT_FALSE(structureNameClashingWithJava.getLongField().isValueUsed());
        ASSERT_FALSE(structureNameClashingWithJava.getBigIntegerField().isValueUsed());
        ASSERT_FALSE(structureNameClashingWithJava.getFloatField().isValueUsed());
        ASSERT_FALSE(structureNameClashingWithJava.getDoubleField().isValueUsed());
        ASSERT_FALSE(structureNameClashingWithJava.getStringField().isValueUsed());
    }

    {
        StructureNameClashingWithJava structureNameClashingWithJava = {};
        ASSERT_FALSE(structureNameClashingWithJava.getByteField().isValueUsed());
        ASSERT_FALSE(structureNameClashingWithJava.getShortField().isValueUsed());
        ASSERT_FALSE(structureNameClashingWithJava.getIntegerField().isValueUsed());
        ASSERT_FALSE(structureNameClashingWithJava.getLongField().isValueUsed());
        ASSERT_FALSE(structureNameClashingWithJava.getBigIntegerField().isValueUsed());
        ASSERT_FALSE(structureNameClashingWithJava.getFloatField().isValueUsed());
        ASSERT_FALSE(structureNameClashingWithJava.getDoubleField().isValueUsed());
        ASSERT_FALSE(structureNameClashingWithJava.getStringField().isValueUsed());
    }
}

TEST_F(StructureNameClashingWithJavaTest, bitSizeOf)
{
    StructureNameClashingWithJava structureNameClashingWithJava{Byte{static_cast<int8_t>(0)},
            Short{static_cast<int16_t>(0)}, Integer{static_cast<int32_t>(0)}, Long{static_cast<int64_t>(0)},
            BigInteger{static_cast<uint64_t>(0)}, Float{0.0F}, Double{0.0}, String{""}};

    ASSERT_EQ(BIT_SIZE, structureNameClashingWithJava.bitSizeOf());
}

} // namespace structure_name_clashing_with_java
} // namespace identifiers

File: test/extensions/language/identifiers/java/identifiers/StructureNameClashingWithJavaTest.java
package identifiers;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

import identifiers.structure_name_clashing_with_java.BigInteger;
import identifiers.structure_name_clashing_with_java.Byte;
import identifiers.structure_name_clashing_with_java.Double;
import identifiers.structure_name_clashing_with_java.Float;
import identifiers.structure_name_clashing_with_java.Integer;
import identifiers.structure_name_clashing_with_java.Long;
import identifiers.structure_name_clashing_with_java.Short;
import identifiers.structure_name_clashing_with_java.String;
import identifiers.structure_name_clashing_with_java.StructureNameClashingWithJava;

public class StructureNameClashingWithJavaTest
{
    @Test
    public void emptyConstructor()
    {
        final StructureNameClashingWithJava structureNameClashingWithJava = new StructureNameClashingWithJava();
        assertEquals(null, structureNameClashingWithJava.getByteField());
        assertEquals(null, structureNameClashingWithJava.getShortField());
        assertEquals(null, structureNameClashingWithJava.getIntegerField());
        assertEquals(null, structureNameClashingWithJava.getLongField());
        assertEquals(null, structureNameClashingWithJava.getBigIntegerField());
        assertEquals(null, structureNameClashingWithJava.getFloatField());
        assertEquals(null, structureNameClashingWithJava.getDoubleField());
        assertEquals(null, structureNameClashingWithJava.getStringField());
    }

    @Test
    public void bitSizeOf()
    {
        final StructureNameClashingWithJava structureNameClashingWithJava = new StructureNameClashingWithJava(
                new Byte(java.lang.Byte.valueOf((byte)0)), new Short(java.lang.Short.valueOf((short)0)),
                new Integer(java.lang.Integer.valueOf(0)), new Long(java.lang.Long.valueOf((long)0)),
                new BigInteger(java.math.BigInteger.ZERO), new Float(java.lang.Float.valueOf(0.0f)),
                new Double(java.lang.Double.valueOf(0.0)), new String(""));
        assertEquals(BIT_SIZE, structureNameClashingWithJava.bitSizeOf());
    }

    private static final int BIT_SIZE = 8 * 1 + // all auto optionals
            8 + // Byte
            16 + // Short
            32 + // Integer
            64 + // Long
            64 + // BigInteger
            32 + // Float
            64 + // Double
            8; // String '\0'
}
